java/semantickernel-connectors-parent/semantickernel-connectors/src/main/java/com/microsoft/semantickernel/connectors/memory/sqlite/SqliteMemoryStore.java

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Microsoft. All rights reserved.
+package com.microsoft.semantickernel.connectors.memory.sqlite;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.semantickernel.ai.embeddings.Embedding;
+import com.microsoft.semantickernel.memory.MemoryRecord;
+import com.microsoft.semantickernel.memory.MemoryStore;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.util.function.Tuple2;
+
+import java.sql.*;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+
+public class SqliteMemoryStore implements MemoryStore {
+
+    private final Database dbConnector;
+    private Connection dbConnection;
+
+    public SqliteMemoryStore() {
+        this.dbConnector = new Database();
+    }
+
+    public Mono<Void> connectAsync(String filename) throws SQLException {
```

Comment:
should probably have this called in the builder so the store is connected to a db when created